src/ccall.cpp

```diff
@@ -1573,6 +1573,19 @@ static jl_cgval_t emit_ccall(jl_codectx_t &ctx, jl_value_t **args, size_t nargs)
             tbaa_decorate(tbaa_const, ctx.builder.CreateLoad(ptid)),
             retboxed, rt, unionall, static_rt);
     }
+    else if (is_libjulia_func(jl_get_current_task)) {
+      assert(lrt == T_prjlvalue);
+      assert(!isVa && !llvmcall && nccallargs == 0);
+      JL_GC_POP();
+      Value *ptls_ptr = emit_bitcast(ctx, ctx.ptlsStates, T_pint8);
+      const int task_offset = offsetof(jl_tls_states_t, current_task);
+      Value *pptaski8 = ctx.builder.CreateGEP(
+          ptls_ptr, ConstantInt::get(T_size, task_offset));
+      Value *pptask = emit_bitcast(ctx, pptaski8, T_pprjlvalue);
+      Value *ptask = tbaa_decorate(tbaa_const, ctx.builder.CreateLoad(pptask));
```

Comment:
```
Value *ptask = tbaa_decorate(tbaa_const, ctx.builder.CreateLoad(pptask));
```
I'd be especially grateful if one of the more core people could confirm that `tbaa_const` is right/save here.